src/NuGet.Clients/NuGet.PackageManagement.UI/PackageLicenseUtilities.cs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using NuGet.Packaging;
+using NuGet.Packaging.Licenses;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.PackageManagement.UI
+{
+    internal class PackageLicenseUtilities
+    {
+        private static string LicenseFormat = "https://licenses.nuget.org/licenses/{0}";
+
+        internal static IReadOnlyList<IText> GenerateLicenseLinks(DetailedPackageMetadata metadata)
+        {
+            return GenerateLicenseLinks(metadata.LicenseMetadata, metadata.LicenseUrl);
+        }
+
+        internal static IReadOnlyList<IText> GenerateLicenseLinks(IPackageSearchMetadata metadata)
+        {
+            return GenerateLicenseLinks(metadata.LicenseMetadata, metadata.LicenseUrl);
+        }
+
+        internal static IReadOnlyList<IText> GenerateLicenseLinks(LicenseMetadata licenseMetadata, Uri licenseUrl)
+        {
+            if (licenseMetadata != null)
+            {
+                return GenerateLicenseLinks(licenseMetadata);
+            }
+            else if (licenseUrl != null)
+            {
+                return new List<IText>() { new LicenseText(Resources.Text_LicenseAcceptance, licenseUrl) };
+            }
+            return new List<IText>();
+        }
+
+        // Internal for testing purposes.
+        internal static IReadOnlyList<IText> GenerateLicenseLinks(LicenseMetadata metadata)
+        {
+            var list = new List<IText>();
+            switch (metadata.Type)
+            {
+                case LicenseType.Expression:
+
+                    if (metadata.LicenseExpression != null)
+                    {
+                        IList<string> identifiers = new List<string>();
+                        PopulateLicenseIdentifiers(metadata.LicenseExpression, ref identifiers);
```

Comment:
if you are only adding you don't need to have the param as `ref`

Reply:
good catch.
I think initially I was assigning that reference inside.